src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1154,519 @@ struct Wallet
+    * param: ptx -
+    * param: tx_device_aux -
+    * note: sets status error on fail
+    */
+    virtual void coldTxAuxImport(const PendingTransaction &ptx, const std::vector<std::string> &tx_device_aux) const = 0;
+// TODO : wallet2::signed_tx_set & cryptonote::address_parse_info
+    /**
+    * brief: coldSignTx -
+    * param: ptx -
+    * param: exported_txs -
+    * param: dsts_info -
+    * param: tx_device_aux -
+    */
+//    virtual void coldSignTx(const PendingTransaction &ptx, signed_tx_set &exported_txs, std::vector<cryptonote::address_parse_info> &dsts_info, std::vector<std::string> & tx_device_aux) const = 0;
+    // TODO : this already exists, remove after checking
+    /**
+    * brief: useForkRules - check if the wallet uses the rules for hard fork version at current block height - early_blocks
+    * param: version - hard fork version
+    * param: early_blocks -
+    * return: true if fork rules are used
+    * note: sets status error on fail
+    */
+//    virtual bool useForkRules(std::uint8_t version, std::int64_t early_blocks) const = 0;
+// TODO : wallet2::transfer_details
+    /**
+    * brief: getTransferDetails -
+    * param: idx - index in transfer storage
+    * return: transfer details for given index
+    * note: sets status error on fail
+    */
+//    virtual const wallet2::transfer_details &getTransferDetails(std::size_t idx) const = 0;
+    /**
+    * brief: discardUnmixableOutputs - freeze all unmixable outputs
+    * note: sets status error on fail
+    */
+    virtual void discardUnmixableOutputs() = 0;
+    /**
+    * brief: setTxKey - set the transaction key (r) for a given <txid> in case the tx was made by some other device or 3rd party wallet
+    * param: txid -
+    * param: tx_key - secret transaction key r
+    * param: additional_tx_keys -
+    * param: single_destination_subaddress -
+    * note: sets status error on fail
+    */
+    // QUESTION : There were some cases already where I wished to have boost::optional in here. I have no explicit example, but in general e.g. for functions that return a bool, but have to return early on error, it would be cleaner to return boost::none instead of true/false, because the API user could forget to check the error status and may use an incorrect value. Similar for uint*_t, where we can't use e.g. -1 on error.
+    //            If we add access to boost::optional revisit all new functions to see which would benefit from it.
+    virtual void setTxKey(const std::string &txid, const std::string &tx_key, const std::vector<std::string> &additional_tx_keys, const boost::optional<std::string> &single_destination_subaddress) = 0;
+    /**
+    * brief: getDaemonAddress -
+    * return: daemon address
+    */
+    virtual std::string getDaemonAddress() const = 0;
+    /**
+    * brief: getDaemonAdjustedTime -
+    * return: daemon adjusted time
+    * note: sets status error on fail
+    */
+    virtual std::uint64_t getDaemonAdjustedTime() const = 0;
```

Comment:
As far as I could see in the code of the CLI wallet that uses this method it's only there to support that super-special case where you could lock a transaction not by giving a blockheight, but a time span, e.g. "10 years". Something nobody ever did in earnest. We anyway removed already all CLI commands to lock. 

I am pretty sure we don't need this.

Reply:
A list like the following? Just needs a better place then in the comments of this PR. So far we have:

# List of public `wallet2` methods that do not get a counterpart in the wallet API

|wallet2 method|source|reason|
|-|-|-|
|```bool unset_ring(const std::vector<crypto::key_image> &key_images)```|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L8840)||
|`bool unset_ring(const crypto::hash &txid)`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L8849)||
|`bool find_and_save_rings(bool force = true)`|[source](https://github.com/monero-project/monero/blob/master/src/wallet/wallet2.cpp#L8875)||
|`bool is_output_blackballed(const std::pair<uint64_t, uint64_t> &output) const`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L8970)||
|`bool is_transfer_unlocked(uint64_t unlock_time, uint64_t block_height)`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L7268)||
|`uint64_t get_min_ring_size()`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L8589)|Rings will become deprecated soon with [FCMPs](https://github.com/kayabaNerve/fcmp-ringct/blob/develop/fcmp%2B%2B.pdf).|
|`uint64_t adjust_mixin(uint64_t mixin)`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L8613)|Rings will become deprecated soon with [FCMPs](https://github.com/kayabaNerve/fcmp-ringct/blob/develop/fcmp%2B%2B.pdf).|
|`bool is_deprecated() const`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L1961)|This information is now part of `WalletState`.|
|`std::string get_daemon_address() const`|[source](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/wallet2.cpp#L12898)|This information is now part of `WalletState`.|

Reply:
Yes!